prov/efa/src/rdm/rxr_ep.h

```diff
@@ -279,6 +285,57 @@ static inline int rxr_need_sas_ordering(struct rxr_ep *ep)
 	return ep->msg_order & FI_ORDER_SAS;
 }
 
+static inline uint64_t rxr_get_host_id(char *host_id_file)
+{
+	FILE *fp = NULL;
+	char *host_id_str = NULL;
+	const size_t HOST_ID_LENGTH = 16;
+	const size_t HOST_ID_PREFIX_LENGTH = 3; /* Ignore the i-0 sequence */
+	size_t length = 0;
+	uint64_t host_id = 0;
+
+	if (!host_id_file) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Host id file is not specified\n");
+		goto out;
+	}
+
+	fp = fopen(host_id_file, "r");
+	if (!fp) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot open host id file: %s\n", host_id_file);
+		goto out;
+	}
+
+	fseek(fp, HOST_ID_PREFIX_LENGTH, SEEK_SET);
+
+	host_id_str = calloc(HOST_ID_LENGTH, sizeof(*host_id_str));
+	if (!host_id_str) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot allocate memory\n");
+		goto out;
+	}
+
+	length = fread(host_id_str, 1, HOST_ID_LENGTH, fp);
```

Comment:
`fgets` is better.

Reply:
I double checked the man pages for fread vs fgets but I still think fread is better in this case, since it allows specifying length to read and returns the actual read length. This is very handy for checking the string format.

Reply:
`fgets` handles string better. For example, `fgets` will add `\0` to the end.

Reply:
Thanks I manually added `\0` to the end.